### PR TITLE
Add Working with a Channel guide

### PR DESCRIPTION
**Pull request by b-onc**
The guide is lacking some images, open for suggestions!

### CODE REVIEW

File: stream-chat-swift-docs/docs/guides/working-with-a-channel.md [hunk 1]
```diff
@@ -1,3 +1,298 @@
...
----
\ No newline at end of file
+---
+
+## Getting Channel Data
+
+Here's how you can get data of a Channel you already know the Channel ID of:
+
+```swift
+let channelController = chatClient.channelController(for: ChannelId(type: .messaging, id: <#Your Channel ID#>)
```
Comment (dmigach) [inline stream-chat-swift-docs/docs/guides/working-with-a-channel.md:10]: What we discussed the other day on snippets width applies to almost all the snippets here, they don't fit my MBP16" screen 😢
Screenshot 2021-05-21 at 18 32 38

File: stream-chat-swift-docs/docs/guides/working-with-a-channel.md [hunk 2]
```diff
@@ -1,3 +1,298 @@
...
+Encoded, this will be represented as `messaging:general`.
+For more information about channel types, please refer to [Channel Types](https://getstream.io/chat/docs/ios-swift/channel_features/?language=swift)
+
+## Different Initializers
+
+There are multiple ways of creating a `ChannelController` instance from `ChatClient`. Each way represent a unique usecase, explained below.
+
+Initializing | Use Case | Implications
+-- | -- | --
+`chatClient.channelController(for cid:)` | You already know the `ChannelId` and the Channel is already created on backend | You can directly call actions on this controller without having to call `synchronize`
```
Comment (dmigach) [inline stream-chat-swift-docs/docs/guides/working-with-a-channel.md:55]: Seems like this table's UX could also be improved if we made the first column narrower
Screenshot 2021-05-21 at 18 32 52

File: stream-chat-swift-docs/docs/guides/working-with-a-channel.md [hunk 3]
```diff
@@ -1,3 +1,298 @@
...
+`chatClient.channelController(createChannelWithId:)` | You want to create a new Channel on backend | You should call `synchronize` on the controller before calling any actions or accessing any data.
+`chatClient.channelController(createDirectMessageChannelWith:)` | You want to create a direct message Channel between users. | Direct message Channel ID's are generated by backend. More information is explained below.
+`chatClient.channelController(for query:)` | You want to have a finer control over with data is initally fetched from backend. | Using this initializer, you can control various parameters, explained below.
```
Comment (dmigach) [inline stream-chat-swift-docs/docs/guides/working-with-a-channel.md:58]:
```suggestion
`chatClient.channelController(for query:)` | You want to have a finer control over which data is initally fetched from backend. | Using this initializer, you can control various parameters, explained below.
```
I guess

File: stream-chat-swift-docs/docs/guides/working-with-a-channel.md (hunk 2 above)
Comment (b-onc) [reply stream-chat-swift-docs/docs/guides/working-with-a-channel.md:55]: I've tried a couple of solutions from https://stackoverflow.com/questions/36121672/set-table-column-width-via-markdown/51701842 but none seems to work with Docusaurus... Leaving it for now, to improve later

**Review by dmigach — APPROVED** [review]

**Pull request merged**